parsers/JP.py

```diff
@@ -38,12 +38,45 @@ def fetch_production(zone_key='JP-TK', session=None, target_datetime=None,
     Calculates production from consumption and imports for a given area
     All production is mapped to unknown
     """
+    df = fetch_production_df(zone_key, session, target_datetime)
+    # add a row to production for each entry in the dictionary:
+    
+    datalist = []
+    
+    for i in df.index:
+        data = {
+            'zoneKey': zone_key,
+            'datetime': df.loc[i, 'datetime'].to_pydatetime(),
+            'production': {
+                'biomass': None,
+                'coal': None,
+                'gas': None,
+                'hydro': None,
+                'nuclear': None,
+                'oil': None,
+                'solar': df.loc[i, 'solar'] if 'solar' in df.columns else None,
+                'wind': None,
+                'geothermal': None,
+                'unknown': df.loc[i, 'unknown']
+            },
+            'storage': {},
+            'source': ['occtonet.or.jp', sources[zone_key]]
```

Comment:
can you write as a string instead of an array? Database schema will enforce this in the future